packages/react/hover-card/src/HoverCard.tsx

```diff
@@ -0,0 +1,222 @@
+import * as React from 'react';
+import { composeEventHandlers } from '@radix-ui/primitive';
+import { createContext } from '@radix-ui/react-context';
+import { useControllableState } from '@radix-ui/react-use-controllable-state';
+import * as PopperPrimitive from '@radix-ui/react-popper';
+import { Portal } from '@radix-ui/react-portal';
+import { Presence } from '@radix-ui/react-presence';
+import { Primitive, extendPrimitive } from '@radix-ui/react-primitive';
+import type * as Polymorphic from '@radix-ui/react-polymorphic';
+
+/* -------------------------------------------------------------------------------------------------
+ * HoverCard
+ * -----------------------------------------------------------------------------------------------*/
+
+const HOVERCARD_NAME = 'HoverCard';
+
+type HoverCardContextValue = {
+  open: boolean;
+  onOpenChange(open: boolean): void;
+  onOpen(): void;
+  onClose(): void;
+};
+
+const [HoverCardProvider, useHoverCardContext] = createContext<HoverCardContextValue>(
+  HOVERCARD_NAME
+);
+
+type HoverCardOwnProps = {
+  open?: boolean;
+  defaultOpen?: boolean;
+  onOpenChange?: (open: boolean) => void;
+  openDelay?: number;
+  closeDelay?: number;
+};
+
+const HoverCard: React.FC<HoverCardOwnProps> = (props) => {
+  const {
+    children,
+    open: openProp,
+    defaultOpen,
+    onOpenChange,
+    openDelay = 700,
+    closeDelay = 300,
+  } = props;
+  const openTimerRef = React.useRef(0);
+  const closeTimerRef = React.useRef(0);
+
+  const [open = false, setOpen] = useControllableState({
+    prop: openProp,
+    defaultProp: defaultOpen,
+    onChange: onOpenChange,
+  });
+
+  const handleOpen = React.useCallback(() => {
+    clearTimeout(closeTimerRef.current);
+
+    openTimerRef.current = window.setTimeout(() => {
+      setOpen(true);
+    }, openDelay);
+  }, [openDelay, setOpen]);
+
+  const handleClose = React.useCallback(() => {
+    clearTimeout(openTimerRef.current);
+
+    closeTimerRef.current = window.setTimeout(() => {
```

Comment:
quick nit if I may ask, can we remove the curly brackets around `setOpen` to clean things up even more?